### PR TITLE
Correct erroneous negative reccomendation in PageCache module

### PR DESCRIPTION
**Pull request by davidalger**

### CODE REVIEW

**Comment by orlangur** [comment]
Why you think Varnish is not the only cache setting recommended for production use?

**Comment by mttjohnson** [comment]
It is one thing to encourage or recommend Varnish as an improved implementation for production, and rightfully so. 

It is quite another thing to discourage and not recommend something for production use as it implies there are serious issues with use in production environments. It **implies** that if you aren't using Varnish in production that you should have full page cache disabled entirely because built-in is not suitable for production.

I would expect the recommendation is to have full page cache enabled in production environments and utilize the built-in full page cache instead of having full page cache disabled, and that if it is possible to implement Varnish, that Varnish would be a further improvement over built-in for production use.

Are there serious problems with using the built-in full page cache in production environments, or is it just less optimal?

**Comment by dougkm** [comment]
That is a very valid point.  There are no known serious issues with the built in Page Cache.  

There is a strong recommendation to use Varnish however, if that is not possible the Page Cache provides a viable alternative.

**Comment by dougkm** [comment]
That is a very valid point.

There is a strong recommendation to use Varnish however there are no known serious issues with the built in Page Cache

**Comment by melnikovi** [comment]
Internal ticket created MAGETWO-46445

**Comment by davidalger** [comment]
@orlangur The notion that varnish is the only valid full page cache option for production seems very odd to me considering this was both built into the M2 application and similar technique has been successfully used to scale Magento EE 1.x. Yes, varnish is a reverse proxy completely detached from the application and will scale much better than any self-contained full page caching system can, simply because it means no hits to the application services such as nginx or apache. Many merchants don't need the level of scalability which varnish can provider, but will still benefit dramatically from the full page cache being enabled during what qualifies as peak traffic for them. Particularly with how the M2 page cache can serve a request to a new visitor, just having the single most trafficked page served from the built-in page cache is a huge performance benefit which will result in better conversions because _it will_ be faster than without any page cache. Using file-system or database as a cache backend, yeah, I agree that the built-in page cache would be far less than optimal. But redis may be setup and used as the cache backend to provide memory backed cache storage for the page cache, without imposing the hefty requirement of setting up, configuring and running varnish in front of the application. IMO information on both page cache options should be present in devdocs, with recommendations to use varnish for highly-scalable usage scenarios and redis + built-in page cache for production scenarios with lower scaling requirements.

TL:DR; I believe having this comment in the admin is misleading, and will result in many merchants simply disabling page cache because they find this comment and don't have the ability to setup varnish.

**Comment by johnbendi** [comment]
```
                                                                                  Anybody who has the ability to set up Magento has the ability to set up Varnish. I don't think difficulty should be a question in this context. Though having a choice might be. Actually IME Magento is harder to set up than Varnish.
```

**Comment by davidalger** [comment]
@johnbendi I don't think ability is the question. :) There are merchants who simply do not need it, and the built-in would serve them just fine. But ability _is_ definitely challenged for anyone installing on a shared hosting environment. I'm not saying I would deploy a merchant on such an environment, but there are a lot of merchants out there who can't afford and/or justify the dedicated environment which would typically be needed to deploy varnish.

**Comment by alankent** [comment]
As Doug said, we should fix things to recommend Varnish over the built in cache, but the built in cache is suitable in some circumstances (such as shared hosting where you cannot create separate processes). We should get the wording correct.

**Comment by melnikovi** [comment]
Thank you for the contribution @davidalger. I confirmed with product owner labels for the options "Built-in Cache" and "Varnish Cache (Recommended)". Could you please update pull request?

**Comment by davidalger** [comment]
@melnikovi I've updated the PR and think the added note on the varnish cache label is great.

**Comment by melnikovi** [comment]
@davidalger please update option labels to fully match "Built-in Cache" and "Varnish Cache (Recommended)". Thank you

**Comment by davidalger** [comment]
@melnikovi Apparently I wasn't paying attention the first time… :/ Fixed that!

**Comment by melnikovi** [comment]
@davidalger your pull request accepted. Thank you for your contribution
